lib/buildkite/test_collector/ci.rb

```diff
@@ -69,17 +71,19 @@ def github_actions
       "branch" => ENV["GITHUB_REF_NAME"],
       "commit_sha" => ENV["GITHUB_SHA"],
       "number" => ENV["GITHUB_RUN_NUMBER"],
+      "build_id" => "#{ENV["GITHUB_RUN_ID"]}-#{ENV["GITHUB_RUN_ATTEMPT"]}"
     }
   end
 
   def circleci
     {
       "CI" => "circleci",
-      "key" => "#{ENV["CIRCLE_WORKFLOW_ID"]}-#{ENV["CIRCLE_BUILD_NUM"]}",
+      "key" => ENV["CIRCLE_WORKFLOW_ID"],
       "url" => ENV["CIRCLE_BUILD_URL"],
       "branch" => ENV["CIRCLE_BRANCH"],
       "commit_sha" => ENV["CIRCLE_SHA1"],
-      "number" => ENV["CIRCLE_BUILD_NUM"],
+      "number" => ENV["CIRCLE_WORKFLOW_ID"], # There is no human readable ENV in CircleCi
```

Comment:
I see that `CIRCLE_WORKFLOW_ID` is a string, would this create any issues with the cast we are doing in the query for `find_build_ids`? Could this string be a UUID rather than just a count that happens in BK?

Reply:
Might also mean that the `#` we are appending to the number in the UI might look a bit weird when it is something like `Build #550e8400-e29b-41d4-a716-446655440000`

Reply:
I'm less worried about the `#` look, as we're constrained by circleci's vars on offer. We could always truncate the build number or something. 

I did double check the cast and you're correct that it breaks, as we can't cast the string uuid to an integer. Having a look for ways around this atm; @blaknite probably has ideas.

Reply:
ChatGPT and I got this working with the ammendment to the find_build_id SQL:
```
SELECT
            env->>'build_id' AS build_id,
            (CASE
              WHEN env->>'number' ~ '^-?\d+$' THEN CAST(env->>'number' AS INTEGER)
              WHEN env->>'number' ~ '^-?\d+(\.\d+)?$' THEN CAST(env->>'number' AS NUMERIC)
              ELSE NULL
            END) AS number,
            created_at
          FROM runs
```
<img width="1920" alt="Screenshot 2023-07-05 at 11 07 44 (2)" src="https://github.com/buildkite/test-collector-ruby/assets/13619812/08c653db-0b23-4555-ad1c-74cdc57a05f6">